tests/ffi/tests.cc

```diff
@@ -841,6 +841,12 @@ extern "C" const char *cxx_run_test() noexcept {
   ASSERT(cstr == "foo");
   ASSERT(other_cstr == "test");
 
+  const char* utf8_literal = u8"Test string";
+  const char16_t* utf16_literal  = u"Test string";
+  rust::String utf8_rstring = utf8_literal;
+  rust::String utf16_rstring = utf16_literal;
+  ASSERT(utf8_rstring == utf16_rstring);
```

Comment:
Do you have any insight into why the test suite failed with MSVC on 29b623436fca88aa8af3b3fd5445dd5733ef48b2?

```console
Assertion failed: `utf8_rstring == utf16_rstring`, tests.cc:848
thread 'test_c_call_r' panicked at 'cxx_run_test()', tests\test.rs:238:5
```

Guessing it was the missing `u8` on the string literal right? Unannotated string literal has implementation-defined system encoding whereas `u8"..."` is specified to UTF-8.

But I see that you also removed a multi-byte code point from the test string in between. Why was that?

Reply:
Yeah the original failure was caused by the missing `u8` but when I started debugging a bit more I saw that multi-byte code points still don't print correctly to the console on MSVC even though the UTF-8 and UTF-16 strings then match. To confirm that the issue is not related to the patch I also tried printing the raw literal to the console and got the same garbage. To me this sounds like some peculiarity with string literals on MSVC rather than an issue with the patch itself so I decided to remove the code point rather than to create the illusion that defining them in literals like that works on MSVC.

P.S. Looking around it seems the issue might not be with MSVC, but rather with the Windows console not accepting UTF-8. Either way I couldn't get it printing correctly and I don't care enough about MSVC at the moment to figure out exactly what the issue is.